docsrc/sasl/authentication_mechanisms.rst

```diff
@@ -10,35 +10,53 @@ Mechanisms
 ANONYMOUS
 ---------
 
-.. todo::
-   Content needed here
+This mechanism does not require the client to authenticate or provide any information.
+
+Defined in :rfc:`2245`
+
+.. _MECH-CRAM-MD5:
 
 CRAM-MD5
 --------
 
-.. todo::
-   Content needed here
+This mechanism avoids sending the users password over the network in plain text by hashing the password with a server provided random value (known as a nonce).
+A disadvantage of this mechanism is that the server must maintain a database of **plaintext passwords** for comparison.
+
+Defined in :rfc:`2195`
+
+`draft-ietf-sasl-crammd5 <https://tools.ietf.org/html/draft-ietf-sasl-crammd5>`_ - The CRAM-MD5 SASL Mechanism (CRAM-MD5)
```

Comment:
Should also note this mech is obsolete. https://datatracker.ietf.org/doc/html/draft-ietf-sasl-crammd5-to-historic-00.html